### PR TITLE
added nullchecks for optional <Conditions> attributes (NPE fix)

### PR DESCRIPTION
**Pull request by fbusselgln**
According to the SAML2 specifications at https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf, the Conditions element is optional and so are the attributes NotBefore and NotOnOrAfter.

Currently if Conditions is included in an assertion, but only NotBefore or (xor!) NotOnOrAfter are provided,  the ZonedDateTime.ofInstant() call causes a NullPointerException.

### CODE REVIEW

**Review by mmoayyed — CHANGES_REQUESTED** [review]
Looks good. 

Please add tests, and switch the target branch to master.

**Comment by fbusselgln** [comment]
Tests added and target switched to master!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,10 +7,10 @@ env:
 on:
     push:
         branches:
-            - master
+            - 5.0.x
```
Comment (mmoayyed) [inline .github/workflows/ci.yml:10]: Reset these back please.

File: documentation/docs/release-notes.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ title: Release notes&#58;
 - Added an option to tolerate missing SAML response `Destination` attribute
 - SAML support: don't add the friendly name if it is the same as the (mapped) name (avoid duplication of values)
 - Improve JWT parsing for nested attributes
+- SAML2 response validation can now disable the validation of `authnInstant` by assigning a zero/negative value to
```
Comment (mmoayyed) [inline documentation/docs/release-notes.md:12]: Should be removed.

File: pac4j-saml/src/main/java/org/pac4j/saml/credentials/SAML2Credentials.java [hunk 3]
```diff
@@ -45,9 +45,22 @@ public SAML2Credentials(final SAMLNameID nameId, final String issuerId,
         this.issuerId = issuerId;
         this.sessionIndex = sessionIndex;
         this.attributes = samlAttributes;
-        this.conditions = new SAMLConditions();
-        this.conditions.setNotBefore(ZonedDateTime.ofInstant(conditions.getNotBefore(), ZoneOffset.UTC));
-        this.conditions.setNotOnOrAfter(ZonedDateTime.ofInstant(conditions.getNotOnOrAfter(), ZoneOffset.UTC));
+        
+		// <Conditions> are optional elements in SAML2 assertions, so if NotBefore or
```
Comment (mmoayyed) [inline pac4j-saml/src/main/java/org/pac4j/saml/credentials/SAML2Credentials.java:49]: Remove inline comments; or switch to javadocs.

File: pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2AuthnResponseValidator.java [hunk 4]
```diff
@@ -681,6 +681,12 @@ Boolean wantsAssertionsSigned(final SAML2MessageContext context) {
 
     private boolean isAuthnInstantValid(final SAML2MessageContext context, final Instant authnInstant) {
         var configContext = context.getConfigurationContext();
-        return isDateValid(authnInstant, configContext.getMaximumAuthenticationLifetime());
+        var maximumAuthenticationLifetime = configContext.getMaximumAuthenticationLifetime();
```
Comment (mmoayyed) [inline pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2AuthnResponseValidator.java:684]: Should be removed.

**Review by mmoayyed — CHANGES_REQUESTED** [review]
You seem to have leftover or misc changes. Please reset and remove the extra stuff.

File: pac4j-saml/src/main/java/org/pac4j/saml/credentials/SAML2Credentials.java (hunk 3 above)
Comment (fbusselgln) [reply pac4j-saml/src/main/java/org/pac4j/saml/credentials/SAML2Credentials.java]: Inline comments removed. The documentation is in the commit messages/pullrequest and not really relevant to any users, so I'd rather not put it into javadoc.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fbusselgln) [reply .github/workflows/ci.yml]: I have dropped the unwanted commits.

File: documentation/docs/release-notes.md (hunk 2 above)
Comment (fbusselgln) [reply documentation/docs/release-notes.md]: I believe this is resolved.

**Comment by fbusselgln** [comment]
Is everything ok now? :)

**Review by mmoayyed — APPROVED** [review]

**Comment by fbusselgln** [comment]
Checkstyle let's the build fail because I have committed tabs instead of whitespaces. Apologies!

**Comment by fbusselgln** [comment]
Thanks for merging. Can you roughly tell me how long it will take for the next version to be released? This is quite a blocker for us.

**Comment by leleuj** [comment]
Last release (v5.0.1) was mid-April so it's time for a new one (v5.1.0). I'll do that at the beginning of next week...

**Comment by leleuj** [comment]
Ping me if you don't see anything coming (in case I forget)

**Comment by fbusselgln** [comment]
Many thanks!

**Pull request merged**